### ISSUE TITLE
Help me to understand! what am I doing wrong?

### ISSUE THREAD

**Issue by kvashninsa** [tverweij/KerioIpBlocker#1]
Hello! I did everything according to the instructions
kerio gateway - 192.168.0.254
Traffic rules - OK
![image](https://user-images.githubusercontent.com/77765845/203447704-868ec14e-a070-41a7-a779-123d81dd9bbc.png)
![image](https://user-images.githubusercontent.com/77765845/203447815-0c140265-5303-4921-aeff-8d57543fb9a2.png)
AbuseIP DB Blocked group - 254.254.254.254
![image](https://user-images.githubusercontent.com/77765845/203447764-1beb1644-1569-4879-adb0-9e4be6c6712b.png)
On a Windows computer, the default gateway is 192.168.0.254
![image](https://user-images.githubusercontent.com/77765845/203448171-752f78cd-2fad-488e-af44-11e862b47f0d.png)
The program is added to the scheduler, launched with parameters
![image](https://user-images.githubusercontent.com/77765845/203448004-1b091d00-fc61-4605-98c1-9b403153a086.png)
![image](https://user-images.githubusercontent.com/77765845/203448954-347c4461-45cf-4d33-b403-da4cb9458464.png)
User, password - 100% correct
API - 100% correct
I capture traffic.
![image](https://user-images.githubusercontent.com/77765845/203448518-91ecd6bd-1e20-4374-a19c-517ff6a8b415.png)
Nothing happens, doesn't work, no API requests, traffic rules don't capture usage.
![image](https://user-images.githubusercontent.com/77765845/203447870-c8325075-08ee-4a98-a776-7658d853c607.png)

**Comment by tverweij** [comment]
The problem is in the parameters. Dont give empty parameters.
So, remove -R: -UM: -PM: -MU:

Empty is not the same as not supplied.

**Comment by tverweij** [comment]
Two other points that might be an issue; form the machine that is running the blocker, you should be able to:
- ping the firewall on the gateway address
- open the admin interface

**Comment by kvashninsa** [comment]
Kerio Control -  IP 192.168.0.254
Machine that is running the blocker - Windows Server, IP address 192.168.0.23, Gateway - 192.168.0.254
Fixed parameters to Machine that is running the blocker.
![image](https://user-images.githubusercontent.com/77765845/203684315-dbc17cc3-586a-4241-8cac-7b8083a2fbf5.png)
ping to 192.168.0.254 - ok
tracert to 192.168.0.254 - ok

admin interface Kerio Control
![image](https://user-images.githubusercontent.com/77765845/203687787-d18ad99a-5f76-4807-95d4-9dd2f5a6f706.png)

I check the connection of KerioIPBlocker.exe to Kerio Control - log Filter:
```
[24/Nov/2022 10:22:22] ALLOW "Kerio Control Administration" packet from Ethernet, proto:TCP, len:40, 192.168.0.23:58957 -> 192.168.0.254:4081, flags:[ FIN ACK ], seq:403922931/1347 ack:94112037, win:8193, tcplen:0
[24/Nov/2022 10:22:22] ALLOW "Kerio Control Administration" packet to Ethernet, proto:TCP, len:477, 192.168.0.254:4081 -> 192.168.0.23:58957, flags:[ ACK PSH ], seq:94112037/606 ack:403922932, win:501, tcplen:437
[24/Nov/2022 10:22:22] ALLOW "Kerio Control Administration" packet to Ethernet, proto:TCP, len:64, 192.168.0.254:4081 -> 192.168.0.23:58957, flags:[ FIN ACK PSH ], seq:94112474/1043 ack:403922932, win:501, tcplen:24
[24/Nov/2022 10:22:22] ALLOW "Kerio Control Administration" packet from Ethernet, proto:TCP, len:40, 192.168.0.23:58957 -> 192.168.0.254:4081, flags:[ RST ACK ], seq:403922932/1348 ack:94112474, win:0, tcplen:0
```
There is a connection

Made a test rule with packet registration
![image](https://user-images.githubusercontent.com/77765845/203688555-ab896d9e-ac94-457e-9d94-12e2ab20e781.png)
I took a bad IP 139.59.176.155 and am trying to connect to it
log Filter:
```
[24/Nov/2022 10:38:49] ALLOW "Test Rule" packet from Ethernet, proto:TCP, len:52, 192.168.0.222:56389 -> 139.59.176.155:22, flags:[ SYN ], seq:1104367914/4294967295 ack:0, win:64240, tcplen:0
[24/Nov/2022 10:38:49] ALLOW "Test Rule" packet to Ethernet 2, proto:TCP, len:52, 109.194.47.219:56389 -> 139.59.176.155:22, flags:[ SYN ], seq:1104367914/4294967295 ack:0, win:64240, tcplen:0
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet from Ethernet 2, proto:TCP, len:52, 139.59.176.155:22 -> 109.194.47.219:56389, flags:[ SYN ACK ], seq:2071481944/4294967295 ack:1104367915, win:64240, tcplen:0
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet to Ethernet, proto:TCP, len:52, 139.59.176.155:22 -> 192.168.0.222:56389, flags:[ SYN ACK ], seq:2071481944/4294967295 ack:1104367915, win:64240, tcplen:0
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet from Ethernet, proto:TCP, len:40, 192.168.0.222:56389 -> 139.59.176.155:22, flags:[ ACK ], seq:1104367915/0 ack:2071481945, win:1026, tcplen:0
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet to Ethernet 2, proto:TCP, len:40, 109.194.47.219:56389 -> 139.59.176.155:22, flags:[ ACK ], seq:1104367915/0 ack:2071481945, win:1026, tcplen:0
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet from Ethernet, proto:TCP, len:68, 192.168.0.222:56389 -> 139.59.176.155:22, flags:[ ACK PSH ], seq:1104367915/0 ack:2071481945, win:1026, tcplen:28
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet to Ethernet 2, proto:TCP, len:68, 109.194.47.219:56389 -> 139.59.176.155:22, flags:[ ACK PSH ], seq:1104367915/0 ack:2071481945, win:1026, tcplen:28
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet from Ethernet, proto:TCP, len:557, 192.168.0.222:56388 -> 142.251.1.113:443, flags:[ ACK PSH ], seq:3089552519/0 ack:1846241631, win:1025, tcplen:517
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet from Ethernet 2, proto:TCP, len:40, 139.59.176.155:22 -> 109.194.47.219:56389, flags:[ ACK ], seq:2071481945/0 ack:1104367943, win:502, tcplen:0
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet to Ethernet, proto:TCP, len:40, 139.59.176.155:22 -> 192.168.0.222:56389, flags:[ ACK ], seq:2071481945/0 ack:1104367943, win:502, tcplen:0
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet from Ethernet 2, proto:TCP, len:81, 139.59.176.155:22 -> 109.194.47.219:56389, flags:[ ACK PSH ], seq:2071481945/0 ack:1104367943, win:502, tcplen:41
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet to Ethernet, proto:TCP, len:81, 139.59.176.155:22 -> 192.168.0.222:56389, flags:[ ACK PSH ], seq:2071481945/0 ack:1104367943, win:502, tcplen:41
[24/Nov/2022 10:38:50] ALLOW "Test Rule" packet from Ethernet, proto:TCP, len:1296, 192.168.0.222:56389 -> 139.59.176.155:22, flags:[ ACK PSH ], seq:1104367943/28 ack:2071481986, win:1026, tcplen:1256
```

Block detected incoming intrusion IP's, Block detected outgoing intrusion IP's, Block other traffic from internal networks - does not fix the work of the rules. also no API requests.

**Comment by tverweij** [comment]
You can see the connection of the IP Blocker in the Config log:
[23/Nov/2022 16:14:17] admin - session opened for host 10.10.24.22. User-Agent: buildIT IT-Solutions BV/KerioIPBlocker/2.0.3.0.
[23/Nov/2022 16:14:23] admin - session closed for host 10.10.24.22

If you don't see this every 15 seconds, it can not connect to your Kerio instance.

Look in %appdata%, here you should see the file AbuseIPDB_BackListCache.txt - this is the last fetched blacklist. 
If its not there, it can not connect to the AbuseIPDb website, or the specified API key is wrong.

For the connections that are filtered, look in the Connection Log - it's not filtering on packets, only on connections.

**Comment by tverweij** [comment]
That is the support, as far as it goes - I realize now that every screenshot you post is in Russian.
My guide line is: no support for Russia or Belarus.
